### PR TITLE
Add restart cage command

### PR DESCRIPTION
**Pull request by doylemark**
# Why
Adding restart function to CLI

closes spe-789

# How

Will follow up with a PR to add restart polling, need to update infra deployment first.

For now just suggesting to user to observe progress in evervault dashboard

### CODE REVIEW

File: src/cli/restart.rs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    match restart_cage(
+        restart_args.config.as_str(),
+        restart_args.cage_uuid.as_deref(),
+        api_key.as_str(),
+        restart_args.background,
+    )
+    .await
+    {
+        Ok(_) => println!("Cage restart started"),
+        Err(e) => panic!("{e}"),
```
Comment (lfarrel6) [inline src/cli/restart.rs]: This should be an error message for the user, and the exit code from the API Call

File: src/cli/restart.rs [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    };
+
+    exitcode::SOFTWARE
```
Comment (lfarrel6) [inline src/cli/restart.rs]:
```suggestion
    exitcode::OK
```

File: src/cli/restart.rs [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    )
+    .await
+    {
+        Ok(_) => println!(
+            "Cage restart started, progress can be observed in the Evervault dashboard https://app.evervault.com"
+        ),
+        Err(e) => panic!("{e}"),
+    };
+
+    exitcode::SOFTWARE
```
Comment (hanneary) [inline src/cli/restart.rs]: One small think here, the success case should have `exitcode::OK`

**Review by hanneary — COMMENTED** [review]
once quick thing with exit codes

**Review by lfarrel6 — COMMENTED** [review]
lgtm, just 2 changes

**Review by hanneary — APPROVED** [review]

**Pull request merged**